### ISSUE TITLE
BOW for c and x?

### ISSUE THREAD

**Issue by byzhang** [mchen24/iclr2017#4]
First, it's a very interesting work.
I'm reading your paper and it shows you use BoW to represent the doc and the local context.
![image](https://cloud.githubusercontent.com/assets/872149/24225578/7cd77732-0f1e-11e7-95a9-ac9e908d3eb2.png)

But I cannot find where you do BoW in the code. Based on https://github.com/mchen24/iclr2017/blob/master/doc2vecc.c#L461 and https://github.com/mchen24/iclr2017/blob/master/doc2vecc.c#L441, you are using all embeddings of the local/global context?

Would it affect the "Data dependent regularization"?

**Comment by mchen24** [comment]
In the code, I simply go over each word in the document in order, and overwrite each word with certain probability. Therefore, if a word appears multiple times throughout a document, it can be overwritten in some cases, and remained in others.

**Comment by byzhang** [comment]
It looks different with your paper. Did the experiments use the github code?

Thanks,
-B

**Comment by mchen24** [comment]
Yes, I experimented using this code. The derivation on the data-dependent regularization still applies. It will be off by a constant factor. You can try to derive when x = 2 instead of 1.
